### ISSUE TITLE
Cannot find requirements file during install

### ISSUE THREAD

**Issue by Maffooch** [DefectDojo/django-DefectDojo#1216]
**Bug description**
When doing a fresh install in a docker container, setup.bash cannot find a requirements file to read dependencies from. 

**Steps to reproduce**
1. Navigate to /setup
2. Run setup.bash -n
3. Wait for error

**Expected behavior**
Dependencies should be install by pip from a requirements file.

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [ ] Docker
- [x] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: Ubuntu 18.04 in a docker container
 - DefectDojo version: 8c558b8a: Merge pull request #1045 from ptrovatelli/fix-docker-with-override [2019-06-05 10:39:24 -0500]

**Screenshots** (optional)
<img width="978" alt="Screen Shot 2019-06-05 at 5 07 53 PM" src="https://user-images.githubusercontent.com/46459665/58994062-ed809200-87b4-11e9-978a-a6c4d3b4e4fe.png">

**Comment by mtesauro** [comment]
**TLDR:  Installer defaults to the Dojo source being in /opt/dojo/django-DefectDojo - but this can be overridden by env vars**

Now for the patient:  A couple of questions:

Q1: Which branch are you installing from? master, python3 or dev?  That file (requirements-2.txt) only lives in the dev and python3 branches currently.

Q2: Also, what was the path where the Dojo source was cloned into?  

From what you sent, it looks like the Dojo source was in /opt/dojo from the error above.  The installer assumes a default of /opt/dojo/django-DefectDojo/ as the location of the Dojo source but this can be overridden by:

(1) editing the config-var.sh file (not optimal nor suggested)
(2) setting an environmental variable called DOJO_SOURCE and/or INSTALL_ROOT - either at the setup command or when you fire up the docker.  
    DOJO_SOURCE is where the Dojo source lives 
    INSTALL_ROOT is the parent directory for DOJO_SOURCE.

Examples: 

(A) Traditional way with the setup.bash command:

`# INSTALL_ROOT="/opt/example/" DOJO_SOURCE="/opt/example/dojo" ./setup.bash`

(B) Setting the env vars with the docker command:

```
$ docker run -v `pwd`:/opt/example/dojo -e INSTALL_ROOT='/opt/example/' -e DOJO_SOURCE='/opt/example/dojo' -it --name test-dojo-setup ubuntu:18.04 bash
 [now in container]
# /opt/example/dojo/setup/setup.bash
```

Though you'd really want to set the env variable when you build the docker image but you get the idea and I used the above to confirm it works as expected.

NOTE: The above examples assumes the dojo source is at /opt/example/dojo and the docker command cheats to get the dojo source into container by mounting a local directory into it.

**For other install options that can be overridden at install time see**: https://github.com/DefectDojo/django-DefectDojo/blob/dev/setup/scripts/common/config-vars.sh#L50 for info on all the variables that can be overridden at install time.

There's also some semi-complete documentation of setup.bash at https://github.com/DefectDojo/django-DefectDojo/tree/dev/setup (the README.md file)

One other thing to think about: setup.bash assumes there's a parent directory that has the source in a child directory [1]:
```
opt/
`-- dojo
    `-- django-DefectDojo
        |-- CONTRIBUTING.md
          [stuff removed for brevity]
        |-- README.md
        |-- setup
        |   |-- README.MD
        |   |-- mysql.txt
        |   |-- postgresql.txt
        |   |-- requirements-2.txt
        |   |-- requirements-3.txt
        |   |-- scripts
               [more stuff removed for brevity]
        |   |-- setup.bash
[snip]
```
The idea is that that parent directory can be used for things like the virtual env stuff, logs, any helper scripts or whatever.  The installer will ignore stuff in there except for DOJO_SOURCE so that method is 'installer safe'.

BTW, the way I tested the above examples was with:
`$ docker run -v `pwd`:/opt/example/dojo -it --name test-dojo-setup ubuntu:18.04 bash
`

Finally, setup.bash can be called from anywhere on the filesystem as it will "cd" into the directory where setup.bash exists before doing starting the install so this works:

```
$ docker run -v `pwd`:/opt/dojo/django-DefectDojo -it --name test-dojo-setup ubuntu:18.04 bash
  [now in the container]
# pwd
/
# /opt/dojo/django-DefectDojo/setup/setup.bash
  [and the usual install output follows]
```

[1] How I test setup.bash:

```
$ git clone https://github.com/DefectDojo/django-DefectDojo.git example-run
$ cd example-run
$ git checkout dev
$ docker run -v `pwd`:/opt/dojo/django-DefectDojo -it --name test-dojo-setup ubuntu:18.04 bash
 [now inside the container]
# /opt/dojo/django-DefectDojo/setup/setup.bash 

        ____       ____          __     ____          _      
       / __ \___  / __/__  _____/ /_   / __ \____    (_)___  
      / / / / _ \/ /_/ _ \/ ___/ __/  / / / / __ \  / / __ \ 
     / /_/ /  __/ __/  __/ /__/ /_   / /_/ / /_/ / / / /_/ / 
    /_____/\___/_/  \___/\___/\__/  /_____/\____/_/ /\____/  
                                               /___/         

 Welcome to DefectDojo! This is a quick script to get you up and running.
 For more info on how setup.bash does an install, see:
   https://github.com/DefectDojo/django-DefectDojo/tree/master/setup

==============================================================================
  Starting interactive DefectDojo installation
==============================================================================
   [bunch of output removed]
```

**Comment by Maffooch** [comment]
Example B worked like a charm. Excellent work on the educationally thorough explanation!
